### PR TITLE
Fix Scheduler compact appointments rendering on week view

### PR DESCRIPTION
**Pull request by Yarovaya**

### CODE REVIEW

File: js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.base.js [hunk 1]
```diff
@@ -1,36 +1,45 @@
...
+    DROP_DOWN_BUTTON_ADAPTIVE_SIZE = 28;
 
 var BaseRenderingStrategy = Class.inherit({
     ctor: function(instance) {
         this.instance = instance;
+
+        this._initPositioningStrategy();
+    },
+
+    _initPositioningStrategy: function() {
```
Comment (andrewmakarov) [inline js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.base.js:33]: May be 
this._positioningStrategy = this.instance.fire("isAdaptive") ? new AdaptivePositioningStrategy(this) : new BasePositioningStrategy(this);

File: js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.base.js [hunk 2]
```diff
@@ -1,36 +1,41 @@
-var noop = require("../../../core/utils/common").noop,
-    Class = require("../../../core/class"),
-    extend = require("../../../core/utils/extend").extend,
-    errors = require("../../widget/ui.errors"),
-    dateUtils = require("../../../core/utils/date"),
-    isNumeric = require("../../../core/utils/type").isNumeric,
-    typeUtils = require("../../../core/utils/type"),
-    themes = require("../../themes");
+import BasePositioningStrategy from "./ui.scheduler.appointmentsPositioning.strategy.base";
```
Comment (Olexi) [inline js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.base.js:1]: "Base" is often used as a postfix, not suffix. And it is used when it is a template parent component (see TooltipStrategies for example). I suggest that it can be named as GenericPositioningStrategy, of DefaultPositioningStrategy.

File: js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.horizontal.js [hunk 3]
```diff
@@ -14,7 +14,7 @@ var HorizontalRenderingStrategy = BaseAppointmentsStrategy.inherit({
     },
 
     calculateAppointmentWidth: function(appointment, position, isRecurring) {
-        var cellWidth = this._defaultWidth || this.getAppointmentMinSize(),
+        var cellWidth = this.getDefaultCellWidth() || this.getAppointmentMinSize(),
```
Comment (Olexi) [inline js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.horizontal.js:17]: Why did you move getter of _defaultWidth to separate function? Did you override this function in different strategies?

File: js/ui/scheduler/rendering_strategies/ui.scheduler.appointmentsPositioning.strategy.base.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+                allDay: renderingStrategy.instance._groupOrientation === "vertical" ? allDayCount : renderingStrategy.instance.option("_appointmentCountPerCell"),
+                simple: this._calculateDynamicAppointmentCountPerCell() || this._getAppointmentMinCount()
+            };
+        } else {
+            return allDayCount;
+        }
+    }
+
+    getDropDownAppointmentHeight() {
+        return undefined;
```
Comment (Olexi) [inline js/ui/scheduler/rendering_strategies/ui.scheduler.appointmentsPositioning.strategy.base.js:66]: Is `undefined` a standard behavior on function return?

File: js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.base.js (hunk 2 above)
Comment (Yarovaya) [reply js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.base.js:1]: This name ("base") was used because it is consistent with rendering strategies. Rendering strategies are BaseRenderingStrategy, HorizontalRenderingStrategy, VerticalRenderingStrategy and so on.

File: js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.horizontal.js (hunk 3 above)
Comment (Yarovaya) [reply js/ui/scheduler/rendering_strategies/ui.scheduler.appointments.strategy.horizontal.js:17]: I think it is a best practice to avoid addressing fields with underscores. Not only in children, but and in this class too. In other words, I just described a getter for a class field.

File: js/ui/scheduler/rendering_strategies/ui.scheduler.appointmentsPositioning.strategy.base.js (hunk 4 above)
Comment (Yarovaya) [reply js/ui/scheduler/rendering_strategies/ui.scheduler.appointmentsPositioning.strategy.base.js:66]: In this strategy there is no calculated collector height. Further, the height is given in css.

**Review by andrewmakarov — APPROVED** [review]

**Review by Olexi — APPROVED** [review]

**Review by jsdmitry — APPROVED** [review]
